test/com/sun/jna/NativeLibraryTest.java

```diff
@@ -284,6 +288,36 @@ public void testInterceptLastError() {
         assertEquals("Wrong error", EXPECTED, get.invokeInt(null));
     }
 
+    public void testCleanupOnLoadError() throws Exception {
+        Map options = new HashMap();
+        options.put(Library.OPTION_CLASSLOADER, new DisfunctClassLoader());
+        int previousTempFileCount = Native.getTempDir().listFiles().length;
+        try {
+            NativeLibrary.getInstance("disfunct", options);
+            fail("Expected NativeLibrary.getInstance() to fail with an UnsatisfiedLinkError here.");
+        } catch(UnsatisfiedLinkError e) {
+            int currentTempFileCount = Native.getTempDir().listFiles().length;
+            assertEquals("Extracted native library should be cleaned up again. Number of files in temp directory:", previousTempFileCount, currentTempFileCount);
+        }
+    }
+
+    // returns unloadable "shared library" on any input
+    private class DisfunctClassLoader extends ClassLoader {
+        public URL getResource(String name) {
+            try {
+                return new URL("jar", "", name);
+            } catch(MalformedURLException e) {
+                e.printStackTrace();
```

Comment:
I had noticed while writing the test that some detail information was helpful in case of this test setup error. However, I have reduced this information's verbosity from stacktrace to including the exception message in the failure message now.